### ISSUE TITLE
GPS disabled warning but GPS works fine

### ISSUE THREAD

**Issue by ae5960e8-a6fc-491f-b252-898ecf59af95** [OpenTracksApp/OpenTracks#1429]
**Describe the bug**
When I'm recording a track, a message popup appears at the bottom of the screen, disappearing, them appearing again and again. It says GPS is disabled but this issue has been happening for a while and my GPS recordings work with no issue. I'm using a Bluetooth sensor for speed and cadence.

**To Reproduce**
1. Start a track
2. The issue occurs, repeating again and again

If applicable:
![1vauf59bvok39q8b](https://user-images.githubusercontent.com/76442625/203670497-4cae3239-6e13-4f28-9531-84f526851ad5.jpg)

**Technical information**
 - Device: Galaxy Note 10
 - OS: IodeOS
 - OpenTracks version: v4.1.2

**Comment by dennisguse** [comment]
IIRC this message pops when there was not received GPS location for some time (30s?).
So, the message is slightly misleading.

**Comment by ae5960e8-a6fc-491f-b252-898ecf59af95** [comment]
This still happens on the latest version today, GPS works as expected without issue, but the warning pops up every couple of seconds. If this warning could be silenced, I think that would work as a quick fix, as the issue is only the annoying pop-up.

**Comment by nutpantz** [comment]
Same issue on Galaxy s7 except it's in w big red rectangle. Flashes on and off repeatedly. But recording of track seems fine

**Comment by ae5960e8-a6fc-491f-b252-898ecf59af95** [comment]
The message is repeatedly showing up about every second now after I updated to Android 13.

**Comment by nutpantz** [comment]
The gps does not work fine with this bug.
I have tested two  Samsung s7's and a black view 9800. And with the bug there can be over 1500 meter differences between the devices on the latest version from fdroid. When i installed v 3.7 of the app that does not have the bug all 3 devices were within 150 meters for being the exact same.

That is quite a difference. I have tested the new app over ten times and the old app twice.
I will be using the old app and will keep testing if it proves more accurate i will keep to the old version.

**Comment by dennisguse** [comment]
@nutpantz Sounds like a new issue; could you create one?

**Comment by nutpantz** [comment]
I would not say it is a new issue, just that no one has tested the accuracy with this error. After all without the error the accuracy is very very close between two similar devices

**Comment by nutpantz** [comment]
If this fix was in 4.31- 12 gfae on fdroid nightly then the error is still there on my pixel 6

**Comment by dennisguse** [comment]
@nutpantz this issue was about an error Message that was shown without the situation actually occuring. 

What you described seems like a different thing.

**Comment by nutpantz** [comment]
The error message is still showing on the latest nightly i just downloaded. Exactly the same way it was showing before

**Comment by dennisguse** [comment]
@nutpantz created #1532

**Comment by nutpantz** [comment]
Sorry, but the latest nightly 4.33 still has this bug
The gps disabled is flashing with gps accuracy of 8m on s7 edge with linageos Android 11

**Comment by dennisguse** [comment]
@nutpantz could you create a video showing the problem?

**Comment by nutpantz** [comment]
How 
![20230428_095726](https://user-images.githubusercontent.com/12160507/235209017-3fca9fe3-79d4-414b-ad33-def8b3adc13e.jpg)
about a picture

**Comment by dennisguse** [comment]
@nutpantz a video shows me the interval at which these messages are appearing. 
At the moment, I don't know where the error might be. 
A video is remote debugging with style ;)

**Comment by nutpantz** [comment]
That will be tomorrow then

**Comment by nutpantz** [comment]
Latest nightly today

**Comment by dennisguse** [comment]
@nutpantz can you upload it anywhere else and share the link?

**Comment by dennisguse** [comment]
Some more people have problems.
https://github.com/OpenTracksApp/OpenTracks/pull/1514#issuecomment-1551079499

Any suggestions on how to investigate this?

**Comment by macias** [comment]
If I may, there are three things here:

a) IMHO if there is no GPS signal received or GPS is disabled there should be an alarm from OT. The reason being tracing could be important part of work, so if somebody is tracing for a long time, and only after an hour she/he realizes it was for nothing is not too good
b) misleading information -- this is just wording, so it is easy to fix
c) differentiating between GPS disabled and GPS signal lost -- this is quite easy actually. The latter is simple a matter of setting timer, and checking when was the last time GPS was acquired. The former was/is Android event saying there was a switch in GPS provider, so it is a mere subscribing to this event and checking if changed went to `false` or `true`. I've done it in C#/Xamarin using Gingberbread API and it works up to Android 13 :-). But current API could change, my knowledge is rusty

**Comment by nutpantz** [comment]
The real question is if open tracks is recording any data while it thinks the gps is disabled. Is it affecting total distance or any other datum?
Maybe a error log could be exported, with all possible data to help diagnose any issue . then users can easily provide as much info as possible, even when users don't have the ability to get Android logs.

**Comment by dennisguse** [comment]
Can somebody test #1595?
I rewrote the whole status check and added test coverage for this module.

@nutpantz easy the module that did the status checks is likely buggy (or has hidden features), but that does not affect recording of the data :D

**Comment by nutpantz** [comment]
I'm not getting the flashing error on the latest test on two devices
